kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerReadTask.java

```diff
@@ -146,6 +125,8 @@ public boolean doPartialRead() {
             exceededMaxResponseBytes
         );
         finish();
+      } else {
+        parent.finishRead();
```

Comment:
Don't we need this in a `finally`? Otherwise we are still holding the lock if there is an error or we finish successfully. I wonder in fact if we can push the locking into `hasNext()`?

Reply:
No, because calls to `finish()` call finishRead. And `finish()` is called in all other code paths

Reply:
I see. Still, is there any reason not to move the locking into `hasNext`? That keeps the locking encapsulated. The `startRead` and `finishRead` approach only made sense when we had the read-write lock which we held for the duration of the read.

Reply:
Sorry yeah, you're absolutely right